src/lightning_app/core/flow.py

```diff
@@ -634,3 +636,10 @@ def my_remote_method(self, name):
             lightning my_command_name --args name=my_own_name
         """
         raise NotImplementedError
+
+    def on_save_state_dict(self) -> Optional[Dict[str, Any]]:
+        """Override to return any required metadata for reloading."""
+        return None
```

Comment:
@tchaton it is not anymore what we discussed. We can't name it `on_x` when it is required to return state. We need to call it something special, like `extra_state` or whatever to emphasize this is additional data not already captured in the attributes being saved.

Reply:
It's not really `extra_state` IMO, this one needs to return the last resumable state (which may be the last, or not the last, at least not for work, depending on how the specific logic of the work).

I'm wondering what kind of metadata we need to include that we don't consider being in state. As in, state should be defined as what is to resume.